pkg/webhooks/pod/pod.go

```diff
@@ -0,0 +1,174 @@
+
+// GetURI implements Webhook interface
+func (s *PodWebhook) GetURI() string { return "/" + WebhookName }
+
+// SideEffects implements Webhook interface
+func (s *PodWebhook) SideEffects() admissionregv1.SideEffectClass {
+	return admissionregv1.SideEffectClassNone
+}
+
+// Validate implements Webhook interface
+func (s *PodWebhook) Validate(req admissionctl.Request) bool {
+	valid := true
+	valid = valid && (req.UserInfo.Username != "")
+	valid = valid && (req.Kind.Kind == "Pod")
+
+	return valid
+}
+
+func (s *PodWebhook) renderPod(req admissionctl.Request) (*corev1.Pod, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	pod := &corev1.Pod{}
+	if len(req.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(req.OldObject, pod)
+	} else {
+		err = decoder.DecodeRaw(req.Object, pod)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
+
+func (s *PodWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+	pod, err := s.renderPod(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a Pod from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+	//If the incoming Pod is aimed at a privileged namespace except for openshift-logging and openshift-operators, allow it to do whatever it wants.
+	//However, if the pod is targeting a customer's namespace (aka non-privileged), then it may not tolerate certain master/infra node taints.
+	if !privilegedNamespaceRe.Match([]byte(pod.ObjectMeta.GetNamespace())) || exceptionNamespaceRe.Match([]byte(pod.ObjectMeta.GetNamespace())) {
+		for _, toleration := range pod.Spec.Tolerations {
+			if toleration.Key == "node-role.kubernetes.io/infra" && toleration.Effect == corev1.TaintEffectNoSchedule {
+				ret = admissionctl.Denied("Not allowed to schedule a pod with NoSchedule taint on infra node")
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+			if toleration.Key == "node-role.kubernetes.io/infra" && toleration.Effect == corev1.TaintEffectPreferNoSchedule {
+				ret = admissionctl.Denied("Not allowed to schedule a pod with PreferNoSchedule taint on infra node")
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+			if toleration.Key == "node-role.kubernetes.io/master" && toleration.Effect == corev1.TaintEffectNoSchedule {
+				ret = admissionctl.Denied("Not allowed to schedule a pod with NoSchedule taint on master node")
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+			if toleration.Key == "node-role.kubernetes.io/master" && toleration.Effect == corev1.TaintEffectPreferNoSchedule {
+				ret = admissionctl.Denied("Not allowed to schedule a pod with PreferNoSchedule taint on master node")
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+		}
+	}
+
+	// Hereafter, all requests are controlled by RBAC
+	ret = admissionctl.Allowed("Allowed to create a Pod in a privileged Namespace because of RBAC")
```

Comment:
```suggestion
	ret = admissionctl.Allowed("Allowed to create Pod because of RBAC")
```

Here it will apply for all namespaces, not just privileged ones.